source/gui/settingsDialogs.py

```diff
@@ -5961,6 +5962,242 @@ def onSave(self):
+		self._updateCurrentSelection()
+
+	def _updateCurrentSelection(self):
+		"""Update the selection"""
+
+		# ZOOM
+		try:
+			# Get current zoom level from magnifier module
+			currentZoom = magnifier.getCurrentZoomLevel()
+
+			# Ensure it's a float
+			currentZoom = float(currentZoom)
+
+			# Find the closest match in our zoom values
+			closestIndex = 0
+			minDifference = abs(self.zoomValues[0] - currentZoom)
+
+			for i, value in enumerate(self.zoomValues):
+				difference = abs(value - currentZoom)
+				if difference < minDifference:
+					minDifference = difference
+					closestIndex = i
+
+			self.defaultZoomList.SetSelection(closestIndex)
+
+		except (ImportError, ValueError, AttributeError, IndexError) as e:
+			# Default to 2.0x if module not available or value not found
+			log.debug(f"Error getting current zoom: {e}")
+			try:
+				defaultIndex = self.zoomValues.index(2.0)
+				self.defaultZoomList.SetSelection(defaultIndex)
+			except ValueError:
+				self.defaultZoomList.SetSelection(0)
+
+		# FILTER
+		try:
+			# Get current filter from magnifier module
+			currentFilter = magnifier.getCurrentFilter()
+
+			# Ensure it's a string
+			currentFilter = str(currentFilter)
+
+			if currentFilter in self.defaultFilterList.GetStrings():
+				self.defaultFilterList.SetSelection(self.defaultFilterList.GetStrings().index(currentFilter))
+			else:
+				self.defaultFilterList.SetSelection(0)
+
+		except (ImportError, ValueError, AttributeError, IndexError) as e:
+			log.debug(f"Error getting current filter: {e}")
+			try:
+				defaultIndex = self.defaultFilterList.GetStrings().index("None")
+				self.defaultFilterList.SetSelection(defaultIndex)
+			except ValueError:
+				self.defaultFilterList.SetSelection(0)
+
+		# FULLSCREEN MODE
+		try:
+			currentFullscreenMode = magnifier.getCurrentFullscreenMode()
+
+			# Ensure it's a string
+			currentFullscreenMode = str(currentFullscreenMode)
+
+			if currentFullscreenMode in self.defaultFullscreenModeList.GetStrings():
+				self.defaultFullscreenModeList.SetSelection(
+					self.defaultFullscreenModeList.GetStrings().index(currentFullscreenMode)
+				)
+			else:
+				self.defaultFullscreenModeList.SetSelection(0)
+
+		except (ImportError, ValueError, AttributeError, IndexError) as e:
+			log.debug(f"Error getting current fullscreen mode: {e}")
+			try:
+				defaultIndex = self.defaultFullscreenModeList.GetStrings().index("center")
+				self.defaultFullscreenModeList.SetSelection(defaultIndex)
+			except ValueError:
+				self.defaultFullscreenModeList.SetSelection(0)
+
+		# KEEP MOUSE CENTERED
+		try:
+			keepMouseCentered = config.conf["magnifier"]["keepMouseCentered"]
+			# Convert string to boolean if necessary
+			if isinstance(keepMouseCentered, str):
+				keepMouseCentered = keepMouseCentered.lower() in ("true", "1", "yes")
+			self.keepMouseCenteredCheckBox.SetValue(bool(keepMouseCentered))
+		except (KeyError, AttributeError):
+			self.keepMouseCenteredCheckBox.SetValue(False)
```

Comment:
similar notes here:
- are these try/excepts necessary? 
- after adding type hints, are any of these type conversions necessary?